### ISSUE TITLE
Jupyter notebook usage

### ISSUE THREAD

**Issue by eyaltrabelsi** [JasonKessler/scattertext#12]
My environment is Macos  I am using python 2.6 chrome browser 0.0.2.6.0.
I am running the following code 

```
import scattertext as st
import spacy
from pprint import pprint
from IPython.display import display, HTML
%matplotlib inline

convention_df = st.SampleCorpora.ConventionData2012.get_data()  
nlp = 
corpus = st.CorpusFromPandas(convention_df, category_col='party', text_col='text',nlp=spacy.en.English()).build()
html = st.produce_scattertext_explorer(corpus,category='democrat',category_name='Democratic',not_category_name='Republican',width_in_pixels=1000,metadata=convention_df['speaker'])
display(HTML(html.encode('utf-8')))
```

and i am receiving 

> Javascript error adding output!
> TypeError: d3.scaleLinear is not a function
> See your browser Javascript console for more details.

I know that i have installed the package correctly because running the bash example will return legit html.

I would love if you can help me out , and even better if you add example using jupyter because this tool can be super usefull

**Comment by JasonKessler** [comment]
Thanks for using Scattertext and filing this report!

That error gets generated if you can't load either https://cdnjs.cloudflare.com/ajax/libs/d3/4.6.0/d3.min.js or https://d3js.org/d3-scale-chromatic.v1.min.js.  Make sure that your browser's able to access https and please try again.

I'm planning to include a way of using custom URLs for these two libraries in an upcoming version.

**Comment by etrabelsi** [comment]
chrome able to access these addresses but still i receive the same errors, So i am not sure this will solve the issue . i am probably using it wrong :\

**Comment by JasonKessler** [comment]
Hmm.  Please try these two things:

1. If your notebook is accessed through http (vs https) you might be getting a mixed content warning (see https://blog.sucuri.net/2016/04/troubleshooting-mixed-content-warnings-https.html).  The easy and bad way to deal with this is adding a `protocol='http'` to your `produce_scattertext_explorer` call. The problem is will lead to a similar error if you try to access your notebook through https.

2. The right way to do this. In the just-released Scattertext 0.0.2.7.1, you can manually specify the paths of d3.js and  d3-scale-chromatic.v1.js. Use the `d3_url` and `d3_scale_chromatic_url` parameters in `produce_scattertext_explorer`.

For example, if "d3.min.js" and "d3-scale-chromatic.v1.min.js" were present in the current
working directory, calling the following code would reference them locally instead of the remote Javascript files. 

```pydocstring
>>> html = st.produce_scattertext_explorer(corpus,
...          category='democrat',
...          category_name='Democratic',
...          not_category_name='Republican',
...          width_in_pixels=1000,
...          metadata=convention_df['speaker'],
...          d3_url='d3.min.js',
...          d3_scale_chromatic_url='d3-scale-chromatic.v1.min.js')
```
